plugins/actions/validate.py

```diff
@@ -0,0 +1,233 @@
+                    notario.validate(host_vars, lvm_osd_scenario, defined_keys=True)
+
+        except Invalid as error:
+            display.vvvv("Notario Failure: %s" % str(error))
+            msg = "[{}] Validation failed for variable: {}".format(host, error.path[0])
+            display.error(msg)
+            reason = "[{}] Reason: {}".format(host, error.reason)
+            try:
+                if "schema is missing" not in error.message:
+                    given = "[{}] Given value for {}: {}".format(host, error.path[0], error.path[1])
+                    display.error(given)
+                else:
+                    given = ""
+                    reason = "[{}] Reason: {}".format(host, error.message)
+            except KeyError:
+                given = ""
+            display.error(reason)
+            result['failed'] = mode == 'strict'
+            result['msg'] = "\n".join([msg, reason, given])
+            result['stderr_lines'] = msg.split('\n')
+
+        return result
+
+# Schemas
+
+
+def osd_objectstore_choices(value):
+    assert value in ['bluestore', 'filestore'], "osd_objectstore must be either 'bluestore' or 'filestore'"
+
+
+def ceph_origin_choices(value):
+    if not notario_store["containerized_deployment"]:
+        assert value in ['repository', 'distro', 'local'], "ceph_origin must be either 'repository', 'distro' or 'local'"
+
+
+def ceph_repository_choices(value):
+    assert value in ['community', 'rhcs', 'dev'], "ceph_repository must be either 'community', 'rhcs' or 'dev'"
+
+
+def ceph_repository_type_choices(value):
+    assert value in ['cdn', 'iso'], "ceph_repository_type must be either 'cdn' or 'iso'"
+
+
+def validate_monitor_options(value):
+    """
+    Either monitor_address, monitor_address_block or monitor_interface must
+    be defined.
+    """
+    monitor_address_given = notario_store["monitor_address"] != "0.0.0.0"
+    monitor_address_block_given = notario_store["monitor_address_block"] != "subnet"
+    monitor_interface_given = notario_store["monitor_interface"] != "interface"
+
+    msg = "Either monitor_address, monitor_address_block or monitor_interface must be provided"
+
+    assert any([monitor_address_given, monitor_address_block_given, monitor_interface_given]), msg
+
+
+def validate_osd_scenarios(value):
+    assert value in ["collocated", "non-collocated", "lvm"], "osd_scenario must be set to 'collocated', 'non-collocated' or 'lvm'"
+
+
+def validate_objectstore(value):
+    assert value in ["filestore", "bluestore"], "objectstore must be set to 'filestore' or 'bluestore'"
+
+
+def validate_lvm_volumes(value):
+    if notario_store['osd_objectstore'] == "filestore":
+        assert isinstance(value, basestring), "lvm_volumes must contain a 'journal' key when the objectstore is 'filestore'"
+
+
+def validate_ceph_stable_release(value):
+    assert value in ['jewel', 'kraken', 'luminous', 'mimic'], "ceph_stable_release must be set to 'jewel', 'kraken', 'lumious' or 'mimic'"
```

Comment:
I think ['jewel', 'kraken', 'luminous', 'mimic'] should be a list somewhere so we can easily reference it.
It's easier than always adding a new release name in here.

Reply:
This is the list, but I could maybe add a constant or something at the top of this file to make it easier to find?

Reply:
I know it's a list, what I meant is that the list could be a constant that's right :), thanks